src/pages/ProxyVotes.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import styled from "styled-components";
+
+import IntlMessage from "../components/IntlMessage";
+import { AppState } from "../reducers";
+import {
+  Member,
+  MemberLookupTable,
+  GENESIS_MEMBER
+} from "../reducers/membersNew";
+
+interface OwnProps {}
+interface StateProps {
+  members: Member[];
+}
+interface DispatchProps {}
+interface Props {
+  memberAndVotes: [[Member, number]];
+}
+
+const ProxyVotesElem = styled.main`
+  max-width: 90vw;
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+`;
+
+// TODO these should be nice thumbnails presented in a component also used by Leaderboard.tsx
+const ProxyVotes: React.StatelessComponent<Props> = ({ memberAndVotes }) => {
+  return (
+    <ProxyVotesElem>
+      <section>
+        <IntlMessage id="votes.votes" tagName="h1" />
+        <IntlMessage id="votes.heading" tagName="p" />
+        <ol>
+          {memberAndVotes.map(memberAndVote => {
+            const [member, votes] = memberAndVote;
+            return (
+              <li key={member.uid}>
+                {member.fullName}: {votes}
+              </li>
+            );
+          })}
+        </ol>
+      </section>
+    </ProxyVotesElem>
+  );
+};
+
+function getAncestors(
+  member: Member,
+  membersByUid: MemberLookupTable,
+  visited: Set<string> = new Set()
```

Comment:
P2: If this ultimately becomes real code for how the voting would work, I'd rather we do this stuff with immutable data structures (`import { Map, Set } from 'immutable';`). I actually reimplemented most of the redux store stuff that way (instead of with these odd `MemberLookupTables` I used before) in `mobile`, but never brought that stuff back to web-app (I plan on just having the redux stuff as a shared library anyway) so when that happens I'd have to reimplement this stuff regardless.